raiden-cli/src/routes/userDeposit.ts

```diff
@@ -0,0 +1,151 @@
+import { Zero } from '@ethersproject/constants';
+import { BigNumber } from 'ethers';
+import type { Request, Response } from 'express';
+import { Router } from 'express';
+
+import type { Cli } from '../types';
+import {
+  isConflictError,
+  isInsuficientFundsError,
+  isInvalidParameterError,
+} from '../utils/validation';
+
+type DepositResponse = { transaction_hash: string };
+type PlanWithdrawResponse = { transaction_hash: string; planned_withdraw_block_number: number };
+type WithdrawResponse = { transaction_hash: string };
+
+const MalformedNumberValue = new Error('A provided number value was not decodable');
+const MalformedRequestFormat = new Error('The provided JSON is in some way malformed');
+const TotalDepositTooLowError = new Error(
+  'The provided total_deposit is not higher than the previous total_deposit',
+);
+
+function parseAsBigNumber(input: unknown): BigNumber {
+  try {
+    return BigNumber.from(input);
+  } catch {
+    throw MalformedNumberValue;
+  }
+}
+
+function getTotalDepositAmount(request: Request): string | undefined {
+  return request.body.total_deposit?.toString();
+}
+
+function getPlannedWithdrawAmount(request: Request): string | undefined {
+  return request.body.planned_withdraw_amount?.toString();
+}
+
+function getWithdrawAmount(request: Request): string | undefined {
+  return request.body.withdraw_amount?.toString();
+}
+
+function shouldDeposit(request: Request): boolean {
+  return (
+    getTotalDepositAmount(request) !== undefined &&
+    getPlannedWithdrawAmount(request) === undefined &&
+    getWithdrawAmount(request) === undefined
+  );
+}
+
+function shouldPlanWithdraw(request: Request): boolean {
+  return (
+    getTotalDepositAmount(request) === undefined &&
+    getPlannedWithdrawAmount(request) !== undefined &&
+    getWithdrawAmount(request) === undefined
+  );
+}
+
+function shouldWithdraw(request: Request): boolean {
+  return (
+    getTotalDepositAmount(request) === undefined &&
+    getPlannedWithdrawAmount(request) === undefined &&
+    getWithdrawAmount(request) !== undefined
+  );
+}
+
+async function deposit(this: Cli, totalDeposit: string): Promise<DepositResponse> {
+  const totalDepositTarget = parseAsBigNumber(totalDeposit);
+  const currentTotalDeposit = await this.raiden.getUDCCapacity();
```

Comment:
Do we expose the current UDC capacity/deposit somewhere? If not, even if not specified by the requirements, I think we could/should, maybe just a `GET` to the main endpoint returning a simple JSON, otherwise it may be hard for the user to calculate how much they should send to this API. I think the WebUI access this directly, but we don´t want to depend on that.

Besides that, I just noticed this may be a problem, the deposit is expected to receive a `totalDeposit`, but we don´t expose that value on the SDK methods, only the capacity, which may be misleading when dealing with total deposit. We may need to expose and adjust to that if we want to be correct and consistent with the current reference implementation/usage (webUI).

Reply:
#2644 exposes a `Raiden.getUDCTotalDeposit` method, which should allow you to expose a `GET` endpoint (possibly informing both `balance` and `total_deposit`) and base `deposit` amount on `total_deposit` instead of capacity, which should be more correct.

Reply:
✔️ 
I also added a get capacity endpoint because why not...